### ISSUE TITLE
Segmentation fault when training DeepLab segmentation model

### ISSUE THREAD

**Issue by monocongo** [tensorflow/models#8137]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): `conda install tensorflow=1`
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6.7

**Describe the current behavior**
I am trying to train the DeepLab model following the directions in this [DeepLab example](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/pascal.md) using the following command:
```
$ python deeplab/train.py \
    --logtostderr \
    --training_number_of_steps=30000 \
    --train_split="train" \
    --model_variant="xception_65" \
    --atrous_rates=6 \
    --atrous_rates=12 \
    --atrous_rates=18 \
    --output_stride=16 \
    --decoder_output_stride=4 \
    --train_crop_size="513,513" \
    --train_batch_size=1 \
    --dataset="basins" \
    --tf_initial_checkpoint=/home/james/deeplab/pretrained/x65-b2u1s2p-d48-2-3x256-sc-cr300k_init.ckpt.data-00000-of-00001 \
    --train_logdir=./deeplab/datasets/basins/exp/train_on_train_set/train \
    --dataset_dir=./deeplab/datasets/basins
```

I get the following output/error:
```
<thousands of useless/confusing warning messages>
...
2020-01-25 16:03:48.084849: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2808000000 Hz
2020-01-25 16:03:48.086476: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x559e7db9e710 executing computations on platform Host. Devices:
2020-01-25 16:03:48.086916: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Starting Session.
INFO:tensorflow:Saving checkpoint to path ./deeplab/datasets/basins/exp/train_on_train_set/train/model.ckpt
INFO:tensorflow:Starting Queues.
Segmentation fault (core dumped)
```
**Describe the expected behavior**
I am hoping that the model will train as advertised.

**Code to reproduce the issue**
Run a DeepLab model training as described in the DeepLab tutorial documentation referenced above.

BTW I have had this happen on two separate machines, both of which are running Ubuntu 18.04. One is a Dell laptop with CPU and the other is an AWS EC2 instance with T4 GPU. On the EC2 instance the TF version installed is 1.15.0 and I also tried using TensorFlow 2.0 but when I did that the code failed immediately with a message indicating that tf.contrib is no longer included in TensorFlow so my assumption is that this code has not been ported to work for TF2. Please advise if there is a known version of TF that this code works with, it appears to be broken in its current state using the versions of TF that I've tried.

Thanks in advance for any insight or suggestions. And/or if there is a more up-to-date semantic segmentation model from TensorFlow other than DeepLab then please point me in the right direction.

**Comment by Papageno2** [comment]
I got the same error working on CentOS7 with v100 GPU, and the tf=1.15.0.

**Comment by Papageno2** [comment]
> I got the same error working on CentOS7 with v100 GPU, and the tf=1.15.0.

I fix this， see this issue
https://github.com/tensorflow/tensorflow/issues/36209#issuecomment-606663564

**Comment by SteveIb** [comment]
Hi,
I just checked your solution @Papageno2
1- I could make tfrecords successfuly

ls /hdd/Deeplab/models/research/deeplab/datasets/PQR/dataset/tfrecord/
train-00000-of-00004.tfrecord trainval-00002-of-00004.tfrecord
train-00001-of-00004.tfrecord trainval-00003-of-00004.tfrecord
train-00002-of-00004.tfrecord val-00000-of-00004.tfrecord
train-00003-of-00004.tfrecord val-00001-of-00004.tfrecord
trainval-00000-of-00004.tfrecord val-00002-of-00004.tfrecord
trainval-00001-of-00004.tfrecord val-00003-of-00004.tfrecord

2- I have two classes one background and other class, I edited data_generator.py L82-90 as the following
I have 176 samples as shown below

_PASCAL_VOC_SEG_INFORMATION = DatasetDescriptor(
splits_to_sizes={
'train': 150,
'train_aug': 100,
'trainval': 176,
'val': 26,
},
num_classes=2,
ignore_label=255,

3- I have a file called SegmentationClass contains the segmented images using LabelMe like this example
8FX1JXR_CV66B002X0Y2560

In JPEGImages I have png images like this
8FX1JXR_CV66B002X512Y2048

in ImageSets/train.txt I have 150 items name of images without extension val.txt 26 item trainval.txt 150 items.

4- The tensorflow version is 1.14.0

5-When I run the local_test.sh it runs smoothly with no errors, just to check if the error from TF versions, but it seems not.

6-My images sizes are 512*512

Finally:
but When Im running the python train command I'm facing the Segmentation fault (core dumped)

python deeplab/train.py
--logtostderr
--training_number_of_steps=30000
--train_split="train"
--model_variant="xception_65"
--atrous_rates=6
--atrous_rates=12
--atrous_rates=18
--output_stride=16
--decoder_output_stride=4
--train_crop_size="513,513"
--train_batch_size=1
--dataset="pascal_voc_seg"
--tf_initial_checkpoint=${hdd/Deeplab/models/research/deeplab/datasets/PQR/dataset/exp/train_on_tranval_set/init_modelsinit_models/deeplabv3_pascal_train_aug/model.ckpt}
--train_logdir=${hdd/Deeplab/models/research/deeplab/datasets/PQR/dataset/exp/train_on_tranval_set/train/}
--dataset_dir=${hdd/Deeplab/models/research/deeplab/datasets/PQR/dataset/tfrecord/} > output.txt 2>&1

The error is still the same, here is the log file(part of it)
INFO:tensorflow:Running local_init_op.
I0429 10:23:24.512071 140539206354688 session_manager.py:500] Running local_init_op.
INFO:tensorflow:Done running local_init_op.
I0429 10:23:24.996261 140539206354688 session_manager.py:502] Done running local_init_op.
INFO:tensorflow:Starting Session.
I0429 10:23:26.720231 140539206354688 learning.py:754] Starting Session.
INFO:tensorflow:Starting Queues.
I0429 10:23:26.721034 140539206354688 learning.py:768] Starting Queues.
Fatal Python error: Segmentation fault

Thread 0x00007fd1d578c700 (most recent call first):
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1429 in _call_tf_sessionrun
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1341 in _run_fn
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1356 in _do_call
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1350 in _do_run
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1173 in _run
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 950 in run
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/contrib/slim/python/slim/learning.py", line 490 in train_step
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/contrib/slim/python/slim/learning.py", line 775 in train
File "deeplab/train.py", line 458 in main
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/absl/app.py", line 250 in _run_main
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/absl/app.py", line 299 in run
File "/home/jvipai/anaconda3/envs/py35/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 40 in run
File "deeplab/train.py", line 464 in

https://github.com/tensorflow/tensorflow/issues/36209#issuecomment-621073290

**Comment by SteveIb** [comment]
@monocongo Did you figure out how to solve it?

**Comment by monocongo** [comment]
@SteveIb no, sorry, I moved on to other projects and haven't had time to follow up on training this model. When it works TensorFlow is a beautiful thing, but usually, it's like chewing rocks and I'll take any excuse to work on something else. This is one of those instances.

**Comment by ntdat017** [comment]
I found problem when my problem when i call vis.py with no tfrecords file in --dataset_dir. There would load model, call to batch 1, and throw error involve "Segmentation fault (core dumped)".

**Comment by AhmadAlhajKarim** [comment]
The error in my case was that --dataset_dir was not pointed correctly to the tfrecord file (wrong path)

**Comment by yuxiazff** [comment]
When my tfrecord file's name was train.tfrecord, I got the same error.    So i change the name to train-00000-of-000001.tfrecord, it run succeed.
